### DISCUSSION TITLE
Has plasmidverify.py been integrated into metaplasmidspades.py?

### DISCUSSION THREAD

**Discussion by LiguanLi** [ablab/spades#752] [Q&A]
Hi there, 

Thanks for your effort on the amazing tools of SPAdes!

I have installed metaplasmidSPAdes from source code. Wondering whether I need to run plasmidverify.py after running metaplasmidspades.py? has the plasmidverify.py been integrated into metaplasmidspades.py?

It seems I need to install plasmidverify.py independently following assembler/src/plasmid_utils/README.md? Could you help clarify?

Thanks a lot in advance!

Best,
Liguan

**Comment by Dmitry-Antipov** [comment]
>Wondering whether I need to run plasmidverify.py after running metaplasmidspades.py?

Yes, you have to. It's not integrated. No installation needed to run, but you need to meet requirements mentioned in assembler/src/plasmid_utils/README.md (prodigal, hmmsearch, Pfam-A database)

**Comment by LiguanLi** [comment]
Thanks a lot Dmitry for quick reply! 
I have finished metaplasmidspades.py and plasmidverify.py on one of my metagenomic dataset. Two questions following the results I got:

1) Is each contig (if verified as plasmid) in the output of metaplasmidspades predicted as one plasmid? or just plasmid-derived segment?

2) There are some warnings after running metaplasmidspades.py as below (para and log files were attached). Will these warnings cause big problem in assembly process?

Thanks very much in advance!

Best,
Liguan
[params.txt](https://github.com/ablab/spades/files/3352132/params.txt)
[spades.log](https://github.com/ablab/spades/files/3352133/spades.log)

======= SPAdes pipeline finished WITH WARNINGS!

=== Error correction and assembling warnings:
 * 1:07:01.212    26G / 40G   WARN    General                 (kmer_coverage_model.cpp   : 327)   Valley value was estimated improperly, reset to 10
 * 1:07:01.216    26G / 40G   WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 10
 * 2:46:35.393    27G / 40G   WARN    General                 (simplification.cpp        : 479)   The determined erroneous connection coverage threshold may be determined improperly
 * 1:16:11.096    12G / 27G   WARN    General                 (kmer_coverage_model.cpp   : 327)   Valley value was estimated improperly, reset to 63
 * 1:16:11.106    12G / 27G   WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 63
 * 1:07:09.552    16G / 30G   WARN    General                 (kmer_coverage_model.cpp   : 218)   Too many erroneous kmers, the estimates might be unreliable
 * 1:07:46.012    16G / 30G   WARN    General                 (kmer_coverage_model.cpp   : 327)   Valley value was estimated improperly, reset to 6
 * 1:07:46.016    16G / 30G   WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 6
 * 1:16:31.218    14G / 28G   WARN    General                 (kmer_coverage_model.cpp   : 327)   Valley value was estimated improperly, reset to 8
 * 1:16:31.223    14G / 28G   WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 8
 * 1:43:22.598    13G / 25G   WARN    General                 (kmer_coverage_model.cpp   : 327)   Valley value was estimated improperly, reset to 51
 * 1:43:22.603    13G / 25G   WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 51
 * 1:22:38.480    13G / 21G   WARN    General                 (kmer_coverage_model.cpp   : 327)   Valley value was estimated improperly, reset to 28
 * 1:22:38.484    13G / 21G   WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 28
